e2e/vegeta_test.go

```diff
@@ -39,75 +38,179 @@ const (
 
 // ===== Vegeta Helper Functions =====
 
-// createRPCTarget
-// • Returns a vegeta.Targeter for the specified RPC method
-func createRPCTarget(
-	gatewayURL string,
-	jsonrpcReq jsonrpc.Request,
-	headers http.Header,
-) vegeta.Targeter {
-	return func(tgt *vegeta.Target) error {
-		body, err := json.Marshal(jsonrpcReq)
-		if err != nil {
-			return err
+// runServiceTest runs the E2E test for a single EVM service in a test case.
+func runServiceTest(
+	t *testing.T,
+	ctx context.Context,
+	targets []vegeta.Target,
+	serviceConfig ServiceConfig,
```

Comment:
Not a blocker, a TODO works too: The last 4 input arguments to this function look very closely related and need to be consolidated into a single input.

Reply:
Done, I've consolidate these args to be part of the `TestService` struct which significantly reduces the # of args passed to the Vegeta test methods.